server.go

```diff
@@ -250,15 +250,16 @@ func main() {
 		r.Body = http.MaxBytesReader(w, r.Body, Config.UploadLimit)
 		r.ParseMultipartForm(limit)
 
-		file, handler, err := r.FormFile("upload")
+		file, _, err := r.FormFile("upload")
 		if err != nil {
 			panic("Error Retrieving the File")
 		}
 		defer file.Close()
 
 		base := r.URL.Query().Get("id")
 
-		parts := strings.Split(handler.Filename, "/")
+		filename := r.Form.Get("upload_fullpath")
```

Comment:
how about adding something like

```
if filename == "" { 
  filename = handler.Filename
}
```

it will support the folder uploading and in the same time it will not break any other clients who may already use this backend

Reply:
fixed